lib/web/controllers/api/session_controller.ex

```diff
@@ -40,6 +41,40 @@ defmodule Web.Api.SessionController do
     |> redirect(to: Routes.session_path(conn, :new))
   end
 
+  def external_login(conn, _opts) do
+    verify_external_login_request(conn)
+  end
+
+  defp verify_external_login_request(conn) do
+    login_secret = conn |> get_req_header("login-secret") |> List.first()
+
+    if login_secret == System.get_env("LOGIN_SECRET") do
+      with user_jwt <- conn |> get_req_header("user-jwt") |> List.first(),
+           {:ok, userinfo = %{"sub" => id_token}} <- verify_user_jwt(user_jwt),
+           {:ok, user} <-
+             Accounts.map_from_login(userinfo, id_token, Security.extract_remote_ip(conn)) do
```

Comment:
I believe this means the remote_ip of the Rails server will be used here. If we have access to the remote_ip of the end user on the Rails side, it should be passed into the request and use that value instead. Knowing the remote IP of the Rails app doesn't help us too much.
```suggestion
           user_remote_ip <- conn |> get_req_header("user-jwt") |> List.first(), # maybe?
           {:ok, userinfo = %{"sub" => id_token}} <- verify_user_jwt(user_jwt),
           {:ok, user} <-
             Accounts.map_from_login(userinfo, id_token, Security.extract_remote_ip(%{remote_ip: user_remote_ip}) do
```